tests/extended/test_trainer_ext.py

```diff
@@ -71,6 +71,17 @@ def require_apex(test_case):
         return test_case
 
 
+# a candidate for testing_utils
```

Comment:
Will you add some test that will need this? If yes, it should definitely go in testing_utils, but for now this is not used anywhere.

Reply:
I added the decorator for future use. I hope to add some BnB tests once I figure out how to test the integration.

Reply:
yes, this will definitely be needed.

adding the test should be quite simple, And you just add `--adamw_bnb_8bit` after copying some training test. Then can enable `--skip_memory_metrics 0` which will give you memory usage at the end of the training - and so taking a snapshot w/o the optimizer, then with it, and checking that the memory usage is smaller.

If we end up with logging the fact that bnb is used the test will also check that it gets logged.

I can help you with that.

The difficult part is that the CI will need the library - I will add it to the OP TODO list to instrument the CI with bnb installation.

Reply:
Not sure if we need the mocking tests, when we can test the real thing. but since you added it let's keep it, especially since it'll take msecs to run.

Reply:
@stas00 I'll try to add a test this week.

Reply:
@stas00 Let me know what you think of the new test.